src/SimpleSMT/Solver/Process.hs

```diff
@@ -92,9 +92,16 @@ with exe args logger todo = do
   return result
 
 -- | Make the solver process into a solver backend.
-toBackend :: SolverProcess -> IO Solver.Backend
-toBackend solver = do
-  response <- (LBS.hGetContents $ getStdout $ process solver) >>= newIORef
-  return $
-    flip Solver.Backend response $ \cmd ->
-      flip LBS.hPutStrLn cmd $ getStdin $ process solver
+toBackend :: SolverProcess -> Solver.Backend
+toBackend solver =
+  Solver.Backend $ \cmd -> do
+    LBS.hPutStrLn (getStdin $ process solver) cmd
+    toLazyByteString <$> (hGetContentsEager mempty $ getStdout $ process solver)
+  where
+    hGetContentsEager :: Builder -> Handle -> IO Builder
```

Comment:
But this is not very robust, is it ? Even if `\n)\n` works as a delimiter for Z3, it might change or not work for other solvers. I guess something I'd be fine with would be to just use `hGetLine` once, assume the solvers' outputs are always given on a single line (which should be a reasonable assumption), and just make a PR against Z3 to add this option.

Reply:
> it might change or not work for other solvers.

That's correct, and it is the reason why it is a strategy that should be used only when Z3 is used as an external process.

> and just make a PR against Z3 to add this option

This would fix it, too.

Reply:
https://github.com/Z3Prover/z3/issues/6461

Reply:
hm, I see. It does look like one is forced to parse the output of some commands until z3 is changed. Note that this could be bearable long term if the commands in question are used sparingly enough.

I wouldn't construct a `SExpr` though since it involves allocations and conversions to String. Reading the bytestring until the parser is happy, and returning that very same bytestring would be fine, I reckon.